### PR TITLE
Chinese translation of YDKJS

### PR DESCRIPTION
**Pull request by JoeHetfield**
**Yes, I promise I've read the [Contributions Guidelines](CONTRIBUTING.md)** (please feel free to remove this line).

### CODE REVIEW

**Comment by getify** [comment]
Are you OK if I do a squash-merge to bring this in?

**Comment by JoeHetfield** [comment]
Yes, no problem. English typo will not affect this translation.

**Comment by JoeHetfield** [comment]
I'll maintain this branch, fix typo , adjust word order etc. and of course follow the changes on the master branch. Few appendixes are in review, will submit within this month.

File: async & performance/ch1.md [hunk 1]
```diff
@@ -1,58 +1,58 @@
-# You Don't Know JS: Async & Performance
-# Chapter 1: Asynchrony: Now & Later
+# 你不懂JS: 异步与性能
+# 第一章: 异步: 现在与稍后
 
-One of the most important and yet often misunderstood parts of programming in a language like JavaScript is how to express and manipulate program behavior spread out over a period of time.
+在像JavaScript这样的语言中最重要但经常被误解的编程技术之一，就是如何表达和操作跨越一段时间的程序行为。
```
Comment (gaowhen) [inline async & performance/ch1.md]: recommend to add a space between English and Chinese.

File: async & performance/ch1.md [hunk 2]
```diff
@@ -84,81 +84,81 @@ var answer = now();
...
+那么，举例来说，当你的JS程序发起一个从服务器取得数据的Ajax请求时，你在一个函数（通常称为回调）中建立好“应答”代码，然后JS引擎就会告诉宿主环境，“嘿，我就要暂时停止执行了，但不管你什么时候完成了这个网络请求，而且你还得到一些数据的话，请 *回来调* 这个函数。”
 
-The browser is then set up to listen for the response from the network, and when it has something to give you, it schedules the callback function to be executed by inserting it into the *event loop*.
+然后浏览器就会为网络的应答设置一个监听器，当它有东西要交给你的时候，它会通过将回调函数插入 *事件轮询* 来安排它的执行。
 
-So what is the *event loop*?
+那么什么是 *事件轮询*？
 
-Let's conceptualize it first through some fake-ish code:
+让我们先通过一些假想代码来对它形成一个概念：
```
Comment (gaowhen) [inline async & performance/ch1.md]: fake-ish code 「伪代码」

File: async & performance/ch1.md [hunk 3]
```diff
@@ -169,29 +169,29 @@ while (true) {
...
 
-**Note:** We mentioned "up until recently" in relation to ES6 changing the nature of where the event loop queue is managed. It's mostly a formal technicality, but ES6 now specifies how the event loop works, which means technically it's within the purview of the JS engine, rather than just the *hosting environment*. One main reason for this change is the introduction of ES6 Promises, which we'll discuss in Chapter 3, because they require the ability to have direct, fine-grained control over scheduling operations on the event loop queue (see the discussion of `setTimeout(..0)` in the "Cooperation" section).
+**注意：** 我们提到了“直到最近”，暗示着ES6改变了事件轮询队列在何处被管理的性质。这主要是一个正式的技术规范，ES6现在明确地指出了事件轮询应当如何工作，这意味着它技术上属于JS引擎应当关心的范畴内，而不仅仅是 *宿主环境*。这么做的一个主要原因是为了引入ES6的Promises（我们将在第三章讨论），因为人们需要有能力对事件轮询队列的排队操作进行直接，细粒度的控制（参见“协作”一节中关于`setTimeout(..0)`的讨论）。
 
-## Parallel Threading
+## 并行线程
 
-It's very common to conflate the terms "async" and "parallel," but they are actually quite different. Remember, async is about the gap between *now* and *later*. But parallel is about things being able to occur simultaneously.
+将“异步”与“并行”两个词经常被混为一谈，但它们实际上是十分不同的。记住，异步是关于 *现在* 与 *稍后* 之间的间隙。但并行是关于可以同时发生的事情。
```
Comment (gaowhen) [inline async & performance/ch1.md]: “异步”与“并行”两个词经常被混为一谈，但它们实际上是完全不同的。

File: async & performance/ch1.md [hunk 4]
```diff
@@ -169,29 +169,29 @@ while (true) {
...
 
-The most common tools for parallel computing are processes and threads. Processes and threads execute independently and may execute simultaneously: on separate processors, or even separate computers, but multiple threads can share the memory of a single process.
+关于并行计算最常见的工具就是进程与线程。进程和线程独立地，可能同时地执行：在不同的处理器上，甚至在不同的计算机上，而多个线程可以共享一个进程的内存资源。
 
-An event loop, by contrast, breaks its work into tasks and executes them in serial, disallowing parallel access and changes to shared memory. Parallelism and "serialism" can coexist in the form of cooperating event loops in separate threads.
+相比之下，一个事件轮询将它的工作打碎成一系列任务并串行地执行它们，不允许并行访问和更改共享的内存。并行与“串行”可能以在不同线程上的事件轮询协作的形式共存。
 
-The interleaving of parallel threads of execution and the interleaving of asynchronous events occur at very different levels of granularity.
+并行线程执行的穿插，与异步事件的穿插发生在完全不同的粒度等级上：
```
Comment (gaowhen) [inline async & performance/ch1.md]: 并行执行的线程的交叉

File: async & performance/ch1.md [hunk 5]
```diff
@@ -200,11 +200,11 @@ function later() {
 }
 ```
 
-While the entire contents of `later()` would be regarded as a single event loop queue entry, when thinking about a thread this code would run on, there's actually perhaps a dozen different low-level operations. For example, `answer = answer * 2` requires first loading the current value of `answer`, then putting `2` somewhere, then performing the multiplication, then taking the result and storing it back into `answer`.
+虽然`later()`的整个内容将被当做一个事件轮询队列的实体，但当考虑到将要执行这段代码的线程时，实际上也许会有许多不同的底层操作。比如，`answer = answer * 2`首先需要读取当前`answer`的值，再把`2`放在某个地方，然后进行乘法计算，最后把结果存回到`answer`。
```
Comment (gaowhen) [inline async & performance/ch1.md]: 尽管 `later()` 会被整个当作事件轮询队列的一个实体

**Comment by gaowhen** [comment]
@JoeHetfield I should thank you for all your hard work first, sincerely.

And if you don't mind, I hope to do some more review work.

**Comment by JoeHetfield** [comment]
@gaowhen Sure, and thank you. After Kyle merge this PR, I think you could further fork that branch. and send your PR back to it. If Kyle grant me the review and merge operations, I'll bring in your fix.

**Comment by gaowhen** [comment]
@JoeHetfield okay, I surely will do it.

**Comment by JoeHetfield** [comment]
@getify Hi, Kyle. My PR haven't been merged yet, is there anything need me to do?

**Comment by getify** [comment]
i need to manually merge and resolve conflicts. will find time to do that soon.

**Comment by JoeHetfield** [comment]
Got it. Please take your time.

**Comment by ac51300696** [comment]
我咋看到中文版呢。。或者帮助翻译

**Comment by JoeHetfield** [comment]
Hi, Kyle. I'm glad to tell you that all the articles and materials in this repo have been translated into Chinese ( with all recent update and merged PR ), and the work of 1st round review had been done. 

My translation of these books has received lots of positive feedback from may Chinese reader. and I'm very  confident to the quality of the translation.

It would be my great honor if my translation was merge to your repo as the first Chinese edition.

File: es6 & beyond/ch8.md [hunk 6]
```diff
@@ -97,21 +83,21 @@ main()
...
+取代`function *main() { ..`声明的，是我们使用`async function main() { ..`形式声明。而取代`yield`一个promise的，是我们`await`promise。运行`main()`函数的调用实际上返回一个我们可以直接监听的promise。这与我们从一个`run(main)`调用中拿回一个promise是等价的。
 
-Do you see the symmetry? `async function` is essentially syntactic sugar for the generators + promises + `run(..)` pattern; under the covers, it operates the same!
+你看到对称性了吗？`async function`实质上是 generators + promises + `run(..)`模式的语法糖；它在底层的操作是相同的！
 
-If you're a C# developer and this `async`/`await` looks familiar, it's because this feature is directly inspired by C#'s feature. It's nice to see language precedence informing convergence!
+如果你是一个C#开发者而且这种`async`/`await`看起来很熟悉，那是因为这种特性就是由C#的特性直接启发的。很高兴看到语言
 
-Babel, Traceur and other transpilers already have early support for the current status of `async function`s, so you can start using them already. However, in the next section "Caveats", we'll see why you perhaps shouldn't jump on that ship quite yet.
+Babel、Traceur 以及其他转译器已经对当前的`async function`状态有了早期支持，所以你已经可以使用它们了。但是，在下一节的“警告”中，我们将看到为什么你也学还不应该上这艘船。
```
Comment (qinyuhang) [inline es6 & beyond/ch8.md]: typo修正 “为什么你也`学`不该上这艘船” -> "为什么你也`许`不该上这艘船"

**Comment by JoeHetfield** [comment]
All conflicts resolved, valid for merge.

**Comment by JoeHetfield** [comment]
Hi, Kyle. Is there any issue about merging? If so, please tell me what I can help.
You know, I've been waiting for it quite a while...

**Comment by getify** [comment]
Sorry that took me so long to get around to reviewing. Looks great, thanks for all your hard work! Merged!!

**Pull request merged**